org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/internal/workspace/ImportingProjectCrossVersionTest.groovy

```diff
@@ -102,6 +103,54 @@ class ImportingProjectCrossVersionTest extends ProjectSynchronizationSpecificati
                 '''
             }
         }
+
+        simpleProjectDir = dir('simple-build') {
+            file 'settings.gradle', '''
+                rootProject.name = 'simple-build'
+            '''
+            file 'build.gradle', '''
+                apply plugin: 'java'
+                description = 'a simple project.'
+                task myTask {}
+            '''
+            dir('src/main/java/pkg') {
+                file 'Main.java', '''
+                    package pkg;
+
+                    public class Main {
+                        public static void main(String[] args) {
+                            System.out.println("Hello, world!");
+                        }
+                    }
+                '''
+            }
+        }
+    }
+
+    @Unroll
+    def "Can import a simple project build with Gradle #distribution.version"(GradleDistribution distribution) {
+        when:
+        importAndWait(simpleProjectDir, distribution)
+
+        then:
+        allProjects().size() == 1
+        numOfGradleErrorMarkers == 0
+        getPlatformLogErrors().size() == 0
+
+        where:
+        distribution << supportedGradleDistributions
+    }
+
+    @Unroll
+    def "Can not import a simple project build with Gradle #distribution.version"(GradleDistribution distribution) {
+        when:
+        def result = getImportAndWaitResult(simpleProjectDir, distribution)
+
+        then:
+        !result.status.isOK()
```

Comment:
Add assertion for the status message to verify a compatibility error.

Reply:
currently I just check the property `eclipse.pde.launch` to know if we are in testing. https://github.com/eclipse/buildship/blob/b3ebed60b964469e446af96af1d1d01be0cf9ef7/org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/DefaultGradleBuild.java#L108-L109

In this way, we can't know it's a comatibility error. I put this check in order to get the real test result without the compatibility check.

If we just remove this check, we will get the test result under the compability limitation, so we can only test "we can limit the version with the matrix", but we can't test the correctness of the matrix itself.

WDYT?

Reply:
We have to be able to test both things:
- the Gradle-Java compatibility matrix itself
- the expected Buildship behavior when an incompatible combination is being used.

I think we should introduce a new `org.eclipse.buildship.integtest.bypassToolingApiCompatibilityChecks`system property:
- set the system propert on the `crossVersionEclipseTest` task: https://github.com/eclipse/buildship/pull/1161/files#diff-94f7e44129e98e0c4f448630d29f25ed8117e1554b6603e3f341200e9d6c2398R22
- Adjust the `EclipseTest` task implementation. `jvmArgs.addAll(testTask.getJvmArgs());` is missing here: https://github.com/eclipse/buildship/blob/master/buildSrc/src/main/groovy/eclipsebuild/testing/EclipseTestExecuter.java#L157